src/ablation.py

```
@@ -565,12 +585,19 @@ def train_fold(
-            ep_auroc = 0.5
 
-        if ep_auroc > best_auroc:
+        # Guard: don't checkpoint if model weights are NaN
+        has_nan_params = any(torch.isnan(p).any() for p in model.parameters())
+        if has_nan_params:
+            print(f"  NaN in model params at epoch {epoch+1}, skipping checkpoint")
+            ep_auroc = float("nan")
+        else:
+            try:
+                ep_auroc = roc_auc_score(y_te, probs)
+            except Exception:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)